### PR TITLE
Stop using `controller … can` syntax in ledger tests

### PR DESCRIPTION
**Pull request by cocreature**
We’re planning to deprecate that in SDK 2.0 to reduce the confusion
around the implicit observer behavior so in preparation for that, this
PR drops the syntax from all ledger tests.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/test-common/src/main/daml/model/Test.daml [hunk 1]
```diff
@@ -278,9 +273,11 @@ template BranchingControllers
     ctrlFalse : Party
   where
   signatory giver
-  controller (if whichCtrl then ctrlTrue else ctrlFalse) can
-    Delete : ()
-      do return ()
+  observer ctrl
+  let ctrl = if whichCtrl then ctrlTrue else ctrlFalse
```
Comment (nickchapman-da) [inline ledger/test-common/src/main/daml/model/Test.daml:277]: Is it syntactically valid to have the `let ctrl =` line above the `observer` line?

**Review by nickchapman-da — APPROVED** [review]
Looks great. Again, I can't see any mistakes.

I had one tiny comment on one example.

Also, did you consider using choice observers instead of template observers for the code transformation?

File: ledger/test-common/src/main/daml/model/Test.daml (hunk 1 above)
Comment (cocreature) [reply ledger/test-common/src/main/daml/model/Test.daml:276]: yes moved it around

**Comment by nickchapman-da** [comment]
> Looks great. Again, I can't see any mistakes.
> 
> I had one tiny comment on one example.
> 
> Also, did you consider using choice observers instead of template observers for the code transformation?

Ignore that. THat of course has a different semantics

**Pull request merged**